service/resource/persistentvolume/state_test.go

```diff
@@ -0,0 +1,142 @@
+package persistentvolume
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_Resource_RecyclePersistentVolume_GetCurrentState(t *testing.T) {
+	testCases := []struct {
+		Obj                             interface{}
+		ExpectedRecyclePersistentVolume interface{}
+	}{
+		{
+			Obj: &apiv1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "TestPersistentVolume",
+					Annotations: map[string]string{},
+				},
+				Status: apiv1.PersistentVolumeStatus{
+					Phase: "Available",
+				},
+			},
+			ExpectedRecyclePersistentVolume: &RecyclePersistentVolume{
+				Name:         "TestPersistentVolume",
+				State:        "Available",
+				RecycleState: recycled,
+			},
+		},
+		{
+			Obj: &apiv1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "TestPersistentVolume",
+					Annotations: map[string]string{
+						"pv-cleaner-operator.giantswarm.io/volume-recycle-state": cleaning,
+					},
+				},
+				Status: apiv1.PersistentVolumeStatus{
+					Phase: "Bound",
+				},
+			},
+			ExpectedRecyclePersistentVolume: &RecyclePersistentVolume{
+				Name:         "TestPersistentVolume",
+				State:        "Bound",
+				RecycleState: cleaning,
+			},
+		},
+		{
+			Obj: &apiv1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "TestPersistentVolume",
+					Annotations: map[string]string{
+						"pv-cleaner-operator.giantswarm.io/volume-recycle-state": recycled,
+					},
+				},
+				Status: apiv1.PersistentVolumeStatus{
+					Phase: "Released",
+				},
+			},
+			ExpectedRecyclePersistentVolume: &RecyclePersistentVolume{
+				Name:         "TestPersistentVolume",
+				State:        "Released",
+				RecycleState: recycled,
+			},
+		},
+	}
+
+	var err error
+	var newResource *Resource
+	{
+		resourceConfig := DefaultConfig()
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	for i, tc := range testCases {
```

Comment:
You can use subtests here, they fit nicely with table-driven testing and allow you to have a separate test case for each of the table entries. See https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks and https://github.com/giantswarm/aws-operator/blob/master/service/resource/cloudformationv2/adapter/auto_scaling_group_test.go for an example.